### ISSUE TITLE
How do I get started? Is this still a working solution?

### ISSUE THREAD

**Issue by globemediaofficial** [typpo/textbelt#172]
I see that much of the code has not been touched in 3-4 years. Is this because it's abandoned or because there hasn't been any need for change? This sounds amazing if it works but I don't see any guide on getting started with self-hosting. Is there a way to use docker or something similar? Did I miss a guide somewhere? I saw a couple of similar discussions but they are closed and unanswered.

**Comment by daluu** [comment]
Should still work in general I believe, unless certain carriers have blocked access to the email gateways.

But finding a working guide to deploy or even a ready to deploy solution, is a different matter. There are several issues filed regarding a guide/walkthrough with no progress as far as I'm aware.

For almost ready to deploy, we have this, but is isn't functional yet: https://github.com/typpo/textbelt/issues/88.

**Comment by globemediaofficial** [comment]
Take a look at this [GitHub repository](https://github.com/hexeth/textbelt-docker). I think it works, will try to test it soon.

**Comment by ryanckulp** [comment]
hey @globemediaofficial i think i have an easier solution to self-deploy, free: 
https://github.com/typpo/textbelt/issues/176

**Comment by globemediaofficial** [comment]
@ryanckulp 
Thanks, this solution worked for me! I do have an issue though. When I try to send a text, I get the `{"success": true}`, but it assumes the carrier is ATT when it's actually Verizon. When I go to my sent mail, obviously there is an error showing that the email could not be sent to `<my-number>@txt.att.net` since ATT is not my carrier. How can I fix this? I remember there being a `carrier` parameter in the `CURL` example but I can't find that now. Even if I get it to work with that, I don't want users to have to select their carrier.

**Comment by ryanckulp** [comment]
@globemediaofficial i'm not exactly sure how everything works, but it seems like Textbelt attempts to text each provided number at all the supported carriers for that country/region. you should see in your Sent folder several attempted texts to ATT, Verizon, etc.

this is also what makes the free version limited; Gmail disables mass sending for spam prevention, so even trying to fire off 5-10 texts in a few seconds could breach that throttled limit.

i'm considering building a simple Ruby gem that does the following:

* attempts to send a text message free (gem installer will provide email credentials)
* if response status is "failed, gem sends a text message via API credentials (gem install to provide a key)

Textbelt Pro is already quite reasonable, better pricing than Twilio anyway. so i'm thinking this type of solution could essentially knock the price down another 20-50%.

would you use something like this?

**Comment by globemediaofficial** [comment]
> would you use something like this?

@ryanckulp 
Right now my use case isn't worth paying for since I'm still in the development stage for my app. However, I would consider using a solution like this once more users join because the texts would need to be more reliable. I finally got the messages that I tried to send but it happened at 6 AM (8 hours after I sent them). I know that's a carrier issue since they decide when the gateway will send the text.
